### PR TITLE
[NETBEANS-5599] PHP 8.1 Support: New in initializers

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-5599

RFC:https://wiki.php.net/rfc/new_in_initializers

(Part 1)
Parser:

- Fix the grammar file
- Add unit tests for the parser

(Part 2)

- Add unit tests for the Formatter, GotoDeclaration, and MarkOccurrences

(Part 3)

- Fix the Code Completion
- Add unit tests for the Code Completion

#### Screenshot

nb-php81-new-in-initializer-syntax

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Could you please review it when you have time? Sorry for the huge change. If there is no problem, let's merge it. Thanks!

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java [hunk 1]
```diff
@@ -360,7 +361,7 @@ public CodeCompletionResult complete(CodeCompletionContext completionContext) {
         switch (context) {
             case DEFAULT_PARAMETER_VALUE:
                 final CaseInsensitivePrefix nameKindPrefix = NameKind.caseInsensitivePrefix(request.prefix);
-                autoCompleteKeywords(completionResult, request, Arrays.asList("array")); //NOI18N
+                autoCompleteKeywords(completionResult, request, Arrays.asList("array", "new")); //NOI18N
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:364]: Just curious - should not we use `PHP_GLOBAL_CONST_KEYWORDS` here?

File: php/php.editor/tools/ASTPHP5Parser.cup [hunk 2]
```diff
@@ -4310,6 +4312,19 @@ common_scalar:scalar
...
+/* PHP 8.1: New in initializer */
+static_scalar_value_with_class_instance ::=
+static_scalar_value:scalar
+{:
+    RESULT = scalar;
+:}
+
+| T_NEW:n class_name_reference:className ctor_arguments:ctor
+{:
+    RESULT = new ClassInstanceCreation(nleft, ctorright, className, ctor);;
```
Comment (tmysik) [inline php/php.editor/tools/ASTPHP5Parser.cup]: Nitpick: typo `;;`

**Review by tmysik — APPROVED** [review]
Looks good to me.

**Comment by tmysik** [comment]
@junichi11 As always, greatly done. Thanks a lot for your work!

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:364]: Yes, you are right. I'll fix it later (maybe with another fix). Thanks!

File: php/php.editor/tools/ASTPHP5Parser.cup (hunk 2 above)
Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Parser.cup]: Oops, Will fix it with the next fix. Thanks!

**Comment by junichi11** [comment]
@tmysik Thank you for your review!

File: php/php.editor/tools/ASTPHP5Parser.cup (hunk 2 above)
Comment (tmysik) [reply php/php.editor/tools/ASTPHP5Parser.cup]: No problem :)

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java (hunk 1 above)
Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:364]: No problem either, it is this way for years :)

**Pull request merged**